File: codes/ruby/hello/entry.rb
#!/usr/local/bin/ruby -w

def fact(i)
	if i <= 1
		1
	else
		i * fact(i-1)
	end
end

if __FILE__ == $0
	if ARGV.length < 1
		print "Please enter an integer value\n"
		exit 0
	end

	n = ARGV[0].to_i
	puts "Fact(" + n.to_s + ")= " + fact(n).to_s
end

File: codes/javascript/hello/entry.js
var args = process.argv;

if (args.length < 3) {
    console.log("Please enter an integer value");
    process.exit();
}

var n = args[2];
console.log("Fact(" + n + ")= " + fact(n) );

function fact(i) {
    if (i <= 0) return 1;
    return i * fact(i-1);
}
